src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs

```diff
@@ -237,5 +234,34 @@ internal bool ShouldAddWorkerConfig(string workerDescriptionLanguage)
             }
             return true;
         }
+
+        internal void AddWorkerConcuurencyOptions(RpcWorkerConfig rpcWorkerConfig)
+        {
+            // Read from host.json
+            WorkerConcurrencyOptions options = new WorkerConcurrencyOptions();
+            IConfigurationSection jobHostSection = _config.GetSection(ConfigurationSectionNames.JobHost);
+            var workerConcurrencySection = jobHostSection.GetSection(ConfigurationSectionNames.WorkerConcurrency);
+            if (workerConcurrencySection.Exists())
+            {
+                workerConcurrencySection.Bind(options);
+
+                if (options.DynamicConcurrencyEnabled)
+                {
+                    if (!string.IsNullOrEmpty(_environment.GetEnvironmentVariable(RpcWorkerConstants.FunctionsWorkerProcessCountSettingName)))
+                    {
+                        options.DynamicConcurrencyEnabled = false;
```

Comment:
Shouldn't DynamicConcurrencyEnabled override any explicit FunctionsWorkerProcessCount value? e.g. an existing customer may have that and other preexisting concurrency settings configured, and we want them to be able to simply flip a single switch and enable/disable WDC. Otherwise, they'll have to go in and remove settings to try out WDC?

Reply:
No, it should be vise versa: if a customer already have FunctionsWorkerProcessCount specified we should not enable dynamic concurrency as the customer already have the number of workers configured. Yes, the customer need to remove FunctionsWorkerProcessCount in order to use WDC.